ripple/src/main/webapp/businessgallery.js

```diff
@@ -1,45 +1,166 @@
                 </div>
-              `;
-              //Add every line of reconstructed HTML to contentStrings array
-              contentStrings.push(content);
+               `;
+
+                // Append newly created card element to the container
+                if (makeElement) {
+                  var galleryElement = document.getElementById("row");
+                  galleryElement.innerHTML = galleryElement.innerHTML + content + "\n";
+                }
+              }
+            }
+        });
+   });
+}
+
+/* Function that dynamically loads the businesses into the gallery, depending on the tag that was searched/selected. 
+   Filters by the walking time in seconds.*/
+function loadSearchResultsTime(time) {
+    db.collection("businesses").get().then((querySnapshot) => {
+        var makeElement = false;
+        querySnapshot.forEach((doc) => {
+            //Check if the city in Firestore matches the city extracted from the user inputted address.
+            //Also checks if the business's tag list contains the tag that the carousel requires.
+            var tag = localStorage.getItem('galleryPageSearchTag');
+            if (doc.data().address != null && doc.data().address[1] == localStorage.getItem('enteredCity')) {
+              makeElement = true;
+              const card = document.createElement('div');
+              card.classList = 'row';
+
+               // Distance matrix calculations
+              // Get user's latitude and longitude from localStorage
+              var userLat = localStorage.getItem('enteredLat');
+              var userLong = localStorage.getItem('enteredLong');
+
+              // Get business latitude and longitude from GeoPoint object in firestore
+              var busLat = doc.data().coordinates.latitude;
+              var busLong = doc.data().coordinates.longitude;
+
+              // Reconstruct as maps LatLng object
+              var origin = new google.maps.LatLng(userLat, userLong);
+              var destination = new google.maps.LatLng(busLat, busLong);
+              
+              // Call distance matrix service
+              var service = new google.maps.DistanceMatrixService();
+              service.getDistanceMatrix(
+                {
+                  origins: [origin],
+                  destinations: [destination],
+                  travelMode: google.maps.TravelMode.WALKING,
+                }, callback);
+              
+              // In callback function, dynamically construct the HTML for the cards in each carousel
+              function callback(response, status) {
+                if (status !== "OK") {
+                    alert("Error with distance matrix");
+                    return;
+                }
+                 // Construct card content by appending HTML strings
+                var walkingTime = response.rows[0].elements[0].duration.text;
+                var walkingDist = response.rows[0].elements[0].distance.text;
+
```

Comment:
This logic is very similar to the businesscarousels.js file. I also have this logic when I generate walking directions for my map using the Directions service. Consider combining map logic?

Reply:
I think we can work in the maps.js file that I created

Reply:
I will look into generalizing the functions